tensorvis/tensorvisio/src/processors/vtkdatasettotensorfield3d.cpp

```diff
@@ -0,0 +1,155 @@
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+
+#include <modules/tensorvisio/processors/vtkdatasettotensorfield3d.h>
+#include <modules/tensorvisbase/util/misc.h>
+
+#include <warn/push>
+#include <warn/ignore/all>
+#include <vtkRectilinearGrid.h>
+#include <vtkStructuredGrid.h>
+#include <vtkStructuredPoints.h>
+#include <vtkDoubleArray.h>
+#include <vtkFloatArray.h>
+#include <vtkPointData.h>
+#include <vtkCellData.h>
+#include <warn/pop>
+
+namespace inviwo {
+
+// The Class Identifier has to be globally unique. Use a reverse DNS naming scheme
+const ProcessorInfo VTKDataSetToTensorField3D::processorInfo_{
+    "org.inviwo.VTKDataSetToTensorField3D",  // Class identifier
+    "VTK Data Set To Tensor Field 3D",       // Display name
+    "VTK",                                   // Category
+    CodeState::Experimental,                 // Code state
+    Tags::CPU,                               // Tags
+};
+const ProcessorInfo VTKDataSetToTensorField3D::getProcessorInfo() const { return processorInfo_; }
+
+VTKDataSetToTensorField3D::VTKDataSetToTensorField3D()
+    : Processor()
+    , dataSetInport_("dataSetInport")
+    , tensorField3DOutport_("tensorField3DOutport")
+    , normalizeExtents_("normalizeExtents", "Normalize extents", false)
+    , arrays_("arrays", "arrays")
+    , offset_{0}
+    , previouslySelectedArrayName_{} {
+    addPort(dataSetInport_);
+    addPort(tensorField3DOutport_);
+
+    addProperty(normalizeExtents_);
+
+    addProperty(arrays_);
+
+    dataSetInport_.onChange([this]() { initializeResources(); });
+}
+
+void VTKDataSetToTensorField3D::initializeResources() {
+    if (!dataSetInport_.hasData()) return;
+
+    const auto& dataSet = *dataSetInport_.getData();
+    auto pointData = dataSet->GetPointData();
+
+    auto stringReplace = [](std::string str, const std::string& from,
+                            const std::string& to) -> const std::string {
+        size_t start_pos{0};
+        while ((start_pos = str.find(from, start_pos)) != std::string::npos) {
+            str.replace(start_pos, from.length(), to);
+            start_pos += to.length();
+        }
+        return str;
+    };
+
+    arrays_.clearOptions();
```

Comment:
consider using `arrays_.replaceOptions(options);`. That will also take care of selecting the same item, if it still exists.

Reply:
Done.